### PR TITLE
Cscmetax 495 512

### PR DESCRIPTION
**Pull request by junsk1**
- Change total byte size fields' data type from decimal to integer
- Drop replication_path field from file model

### CODE REVIEW

**Comment by hannu40k** [comment]
Otherwise OK, but depends on changes in PR https://github.com/CSCfi/metax-api/pull/393

File: src/metax_api/tests/api/rest/base/views/common/write.py [hunk 1]
```diff
@@ -79,6 +79,37 @@ def test_certain_create_fields_are_read_only_after_create(self):
...
+
+        cr_id = response.data['id']
+        response = self.client.delete('/rest/datasets/%d' % cr_id)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT, response.data)
+
+        # Verify date_removed got set
+        response = self.client.get('/rest/datasets/%d?removed' % cr_id)
+        self.assertTrue(response.data['removed'] is True)
+        self.assertTrue(response.data.get('date_removed', False) and
+                        response.data.get('date_removed', '').startswith('2'))
```
Comment (hannu40k) [inline src/metax_api/tests/api/rest/base/views/common/write.py]: Nitpicking, but: This statement: response.data.get('date_removed', False) is redundant I think

File: src/metax_api/api/rest/base/api_schemas/datacatalog.json [hunk 2]
```diff
@@ -59,7 +59,14 @@
...
                     "readonly": true
-                }
+                },
+                "date_removed":{
+                    "title":"Date removed",
+                    "description":"Date of removal of the resource. Can't be updated once set",
+                    "type":"string",
+                    "format":"date-time",
+                    "readonly":true
+                  }
```
Comment (hannu40k) [inline src/metax_api/api/rest/base/api_schemas/datacatalog.json]: Looks like field ``removed`` has been missing from this file too. Good time to add it.

File: src/metax_api/api/rest/base/api_schemas/file.json [hunk 3]
```diff
@@ -248,6 +248,13 @@
                     "description":"Service name who last modified this resource.",
                     "type":"string",
                     "readonly": true
+                },
+                "date_removed":{
+                    "title":"Date removed",
+                    "description":"Date of removal of the resource. Can't be updated once set",
+                    "type":"string",
+                    "format":"date-time",
+                    "readonly":true
```
Comment (hannu40k) [inline src/metax_api/api/rest/base/api_schemas/file.json]: Looks like field ``removed`` has been missing from this file too. Good time to add it.

**Comment by hannu40k** [comment]
In the docs /docs/general.html#retrieving-deleted-objects this chapter should be updated regarding restoring deleted objects (in particular the idea that any update to a deleted object will restore it).

File: src/metax_api/api/rest/base/api_schemas/datacatalog.json (hunk 2 above)
Comment (junsk1) [reply src/metax_api/api/rest/base/api_schemas/datacatalog.json]: Actually that field looks like it's missing from all three api_schemas. I'll add it to them.

**Pull request merged**